providers/linux/boottime_linux.go

```diff
@@ -37,7 +37,7 @@ func bootTime(fs procfs.FS) (time.Time, error) {
 		return bootTimeValue, nil
 	}
 
-	stat, err := fs.NewStat()
+	stat, err := procfs.NewStat()
```

Comment:
this should use fs.Stat, so the base path can be mocked for testing

```suggestion
	stat, err := fs.Stat()
```